src/ansible_builder/ee_schema.py

```diff
@@ -390,16 +391,31 @@
     },
 }
 
+schema_v31: dict = deepcopy(schema_v3)
+schema_v31['properties']['dependencies']['properties']['exclude'] = {
+    "python": TYPE_StringOrListOfStrings,
+    "system": TYPE_StringOrListOfStrings,
+}
+
+
+def float_or_int(v):
+    if isinstance(v, (float, int)):
+        return v
+    if isinstance(v, str):
+        if '.' in v:
+            return float(v)
+    return int(v)
+
 
 def validate_schema(ee_def: dict):
     schema_version = 1
     if 'version' in ee_def:
         try:
-            schema_version = int(ee_def['version'])
+            schema_version = float_or_int(ee_def['version'])
         except ValueError as e:
             raise DefinitionError(f"Schema version not an integer: {ee_def['version']}") from e
 
-    if schema_version not in (1, 2, 3):
+    if schema_version not in (1, 2, 3, 3.1):
```

Comment:
Since we're abusing a combination of other tools in the absence of a first-class YAML "schema" for EE definitions , it's probably better to just support this in the existing version 3 schema, along with a small docs note that the exclude feature requires Builder >= 3.1 (and will be an error on < 3.1).